applications/luci-app-upnp/htdocs/luci-static/resources/view/status/include/80_upnp.js

```diff
@@ -53,12 +51,24 @@ return baseclass.extend({
 		var rules = Array.isArray(data[0].rules) ? data[0].rules : [];
 
 		var rows = rules.map(function(rule) {
+			const padnum = (num, length) => num.toString().padStart(length, "0");
+			const expires_sec = (rule.expires * 1000 - Date.now()) / 1000;
```

Comment:
In the unlikely event that the expires property is unavailable, how about handling it a bit better (so in worst case we get 0):
```js
			const expires_sec = ((rule?.expires * 1000 || Date.now()) - Date.now()) / 1000;
```

Reply:
Although this is where we handle the value in the front end.

Reply:
Feels safer that way. Yes please.